// flowbite-react/lib/cjs/components/Table/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableRow = exports.TableHeadCell = exports.TableHead = exports.TableCell = exports.TableBody = exports.Table = void 0;
var Table_1 = require("./Table");
Object.defineProperty(exports, "Table", { enumerable: true, get: function () { return Table_1.Table; } });
var TableBody_1 = require("./TableBody");
Object.defineProperty(exports, "TableBody", { enumerable: true, get: function () { return TableBody_1.TableBody; } });
var TableCell_1 = require("./TableCell");
Object.defineProperty(exports, "TableCell", { enumerable: true, get: function () { return TableCell_1.TableCell; } });
var TableHead_1 = require("./TableHead");
Object.defineProperty(exports, "TableHead", { enumerable: true, get: function () { return TableHead_1.TableHead; } });
var TableHeadCell_1 = require("./TableHeadCell");
Object.defineProperty(exports, "TableHeadCell", { enumerable: true, get: function () { return TableHeadCell_1.TableHeadCell; } });
var TableRow_1 = require("./TableRow");
Object.defineProperty(exports, "TableRow", { enumerable: true, get: function () { return TableRow_1.TableRow; } });
