// flowbite-react/lib/cjs/components/Footer/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FooterTitle = exports.FooterLinkGroup = exports.FooterLink = exports.FooterIcon = exports.FooterDivider = exports.FooterCopyright = exports.FooterBrand = exports.Footer = void 0;
var Footer_1 = require("./Footer");
Object.defineProperty(exports, "Footer", { enumerable: true, get: function () { return Footer_1.Footer; } });
var FooterBrand_1 = require("./FooterBrand");
Object.defineProperty(exports, "FooterBrand", { enumerable: true, get: function () { return FooterBrand_1.FooterBrand; } });
var FooterCopyright_1 = require("./FooterCopyright");
Object.defineProperty(exports, "FooterCopyright", { enumerable: true, get: function () { return FooterCopyright_1.FooterCopyright; } });
var FooterDivider_1 = require("./FooterDivider");
Object.defineProperty(exports, "FooterDivider", { enumerable: true, get: function () { return FooterDivider_1.FooterDivider; } });
var FooterIcon_1 = require("./FooterIcon");
Object.defineProperty(exports, "FooterIcon", { enumerable: true, get: function () { return FooterIcon_1.FooterIcon; } });
var FooterLink_1 = require("./FooterLink");
Object.defineProperty(exports, "FooterLink", { enumerable: true, get: function () { return FooterLink_1.FooterLink; } });
var FooterLinkGroup_1 = require("./FooterLinkGroup");
Object.defineProperty(exports, "FooterLinkGroup", { enumerable: true, get: function () { return FooterLinkGroup_1.FooterLinkGroup; } });
var FooterTitle_1 = require("./FooterTitle");
Object.defineProperty(exports, "FooterTitle", { enumerable: true, get: function () { return FooterTitle_1.FooterTitle; } });
